script.js

```diff
@@ -1,12 +1,547 @@
+  showStatus.textContent = `Status: ${show.status}`;
+  additionalDetailsDiv.appendChild(showStatus);
+  
+  const showRuntime = document.createElement("p");
+  showRuntime.classList.add("show-runtime");
+  showRuntime.textContent = `Runtime: ${show.runtime} minutes`;
+  additionalDetailsDiv.appendChild(showRuntime);
+  
+
+  // Add click event listener to show name -- back to show button TO BE DONE
+  showName.addEventListener("click", () => {
+    getEpisodes(show.id)
+      .then((episodes) => {
+        allEpisodes = episodes;
+        hideShowListing();
+        showSelectField(allShows);
+        episodeSelectField(episodes);
+        addSearchField();
+        makePageForEpisodes(allEpisodes);
+        updateEpisodeSelect(allEpisodes);
+      })
+      .catch((error) => console.error(error));
+  });
+
+  return showCard;
+}
+
+function hideShowListing() {
+  const showContainer = document.querySelector(".show-container");
+  showContainer.style.display = "none";
+  if (showSearchContainer) {
+    showSearchContainer.style.display = "none";
+  }
+}
+// Function to remove the episode select field
+function removeEpisodeSelectField() {
+  if (episodeSelectContainer && episodeSelectContainer.parentNode) {
+    episodeSelectContainer.parentNode.removeChild(episodeSelectContainer);
+  }
+}
+
+// Function to remove the search field
+function removeSearchField() {
+  if (searchContainer && searchContainer.parentNode) {
+    searchContainer.parentNode.removeChild(searchContainer);
+  }
+  searchContainer = null; 
+}
+
+function hideShowSearchField() {
+  const showSearchContainer = document.querySelector(".show-search-container");
+  if (showSearchContainer) {
+    showSearchContainer.style.display = "none";
+  }
+}
+
+function addSelectOption(selectElement, optionText) {
+  const defaultOption = document.createElement("option");
+  defaultOption.value = "select-show";
+  defaultOption.textContent = optionText;
+  selectElement.appendChild(defaultOption);
 }
 
-function makePageForEpisodes(episodeList) {
+function showSelectField(shows) {
+  const searchSelectDiv = document.querySelector(".search-select");
+
+  const showContainer = document.createElement("div");
+  showContainer.classList.add("show-container");
+
+  showSelect = document.createElement("select");
+  showSelect.classList.add("show-select");
+  showSelect.addEventListener("change", handleShowSelectChange); 
+
+  addSelectOption(showSelect, "Select Show");
+
+  shows.sort((a, b) => a.name.localeCompare(b.name));
+
+  shows.forEach((show) => {
+    const option = document.createElement("option");
+    option.value = show.id;
+    option.textContent = show.name;
+    showSelect.appendChild(option);
+  });
+
+  showContainer.appendChild(showSelect);
+  searchSelectDiv.appendChild(showContainer);
+}
+
+function handleShowSelectChange() {
+  const selectedShowId = showSelect.value;
+  if (selectedShowId !== "select-show") {
+    getEpisodes(selectedShowId)
```

Comment:
Oh I see, interesting division of logic. Is there a particular reason why you've made it this way? You can assign all episodes inside the function getEpisodes avoiding code duplication.